aptos-move/aptos-aggregator/src/aggregator_change_set.rs

```diff
@@ -2,331 +2,388 @@
-        previous_change: &AggregatorChange,
-    ) -> PartialVMResult<()> {
-        match self {
-            // If the current state is Data, then merging with previous state won't change anything.
-            AggregatorChange::Data { .. } | AggregatorChange::StringData { .. } => Ok(()),
-            AggregatorChange::AggregatorDelta {
-                delta,
-                history,
-                max_value,
-            } => match previous_change {
-                AggregatorChange::Data { value: prev_value } => {
-                    let new_data = merge_data_and_delta(*prev_value, delta, history, *max_value)?;
-                    *self = AggregatorChange::Data { value: new_data };
-                    Ok(())
-                },
-                AggregatorChange::StringData { .. } => Err(code_invariant_error(
-                    "Base value cannot be string for an aggregator",
-                )),
-                AggregatorChange::AggregatorDelta {
-                    delta: prev_delta,
-                    max_value: prev_max_value,
-                    history: prev_history,
-                } => {
-                    if max_value != prev_max_value {
-                        return Err(code_invariant_error(
-                            "Max value of aggregator cannot change",
-                        ));
-                    }
-                    let (new_delta, new_history) =
-                        merge_two_deltas(prev_delta, prev_history, delta, history, *max_value)?;
-                    *self = AggregatorChange::AggregatorDelta {
-                        delta: new_delta,
-                        history: new_history,
-                        max_value: *max_value,
-                    };
-                    Ok(())
-                },
-                AggregatorChange::SnapshotDelta { .. } => Err(code_invariant_error(
-                    "SnapshotDelta cannot come before AggregatorDelta",
-                )),
+    pub fn get_apply_base_id(&self, self_id: &AggregatorID) -> ApplyBase {
+        self.get_apply_base_id_option()
+            .unwrap_or(ApplyBase::Previous(*self_id))
+    }
+
+    pub fn apply_to_base(&self, base_value: AggregatorValue) -> PartialVMResult<AggregatorValue> {
+        use AggregatorApplyChange::*;
+
+        Ok(match self {
+            AggregatorDelta { delta } => {
+                AggregatorValue::Aggregator(delta.apply_to(base_value.into_aggregator_value()?)?)
+            },
+            SnapshotDelta { delta, .. } => {
+                AggregatorValue::Snapshot(delta.apply_to(base_value.into_aggregator_value()?)?)
             },
-            AggregatorChange::SnapshotDelta { .. } => {
-                unimplemented!()
+            SnapshotDerived { formula, .. } => {
+                AggregatorValue::Derived(formula.apply(base_value.into_snapshot_value()?))
             },
+        })
+    }
+}
+
+impl AggregatorChange {
+    pub fn get_merge_dependent_id(&self) -> Option<AggregatorID> {
```

Comment:
What does merge_dependent_id mean?

Reply:
for the above and this, more context here :

https://aptos-org.slack.com/archives/C057VPCD7QX/p1695160530418949

will add a comment in code for that